src/prusti/hanoi.ml

```diff
@@ -0,0 +1,30 @@
+
+(* https://github.com/viperproject/prusti-dev/blob/master/prusti-tests/tests/verify_overflow/pass/rosetta/Towers_of_Hanoi_spec.rs *)
+
+let[@pure] valid_pole (x: int): bool =
+  (* Temporary workaround for || *)
+  if x = 1 then true
+  else if x = 2 then true
+  else if x = 3 then true
+  else false
+
+let[@pure] equals (a: int) (b: int): bool = a = b
```

Comment:
Just curious, when do you use `[@pure]` compared to `external`? I was thinking of removing it because the implementation is half-baked without types in the source language.

Reply:
No particular technical reason, I try to use it first and then switch to `external`. It is slightly more convenient